earth_enterprise/src/fusion/gecombineterrain/gecombineterrain.cpp

```diff
@@ -232,28 +234,46 @@ int main(int argc, char **argv) {
                 &khGetopt::RangeValidator<uint32, 0, 1024>);
     options.opt("read_cache_block_size", read_cache_block_size,
                 &khGetopt::RangeValidator<uint32, 1, 1024>);
-
     if (!options.processAll(argc, argv, argn)) {
       usage(progname);
     }
     if (help) {
       usage(progname);
     }
+
     if (argn == argc) {
-      usage(progname, "No input indexes specified");
+      usage(progname, "No input indices specified");
     }
+    uint cmdDefaultCPUs = CommandlineNumCPUsDefault();
+    uint numavailable = GetMaxNumJobs();
+
+    /*
+ 	Choosing number of CPUs:
+
+ 	numcpus: if --numcpu parameter not present, defaults to 1
+	
+	CommandLineNumCPUsDefault(): defined in /etc/opt/google/systemrc via maxjobs tag
+
+	GetMaxNumJobs(): the number of available jobs, defined as the minimum value of either:
+	  1. the currently available jobs or the default number of jobs i.e. kMaxNumJobs (is 8) multiplied by KH_MAX_NUM_JOBS_COEFF, which is a compiler option
+	  2. kMaxNumJobsDefault = 8
+	  3. environment variable, KH_GOOGLE_MAX_NUM_JOBS, if defined
+	  4. the number of currently available cpus, obtained through sysconf
+
+	The methodology	first checks if the --numcpus flag was present, if
+	it was, it will go with what was passed, no matter the number. If
+	it was not, it will take the minimum value of what is defined in the
+	systemrc file and what is currently available (values described 
+	above). If the maxjobs tag in systemrc is invalid (i.e. less than
+	1, it will default to the available CPUs.
+    */
+    if (!numcpus) {
```

Comment:
I'm a bit confused here - your comment above says numcpus defaults to 1 if it's not present, but this seems to indicate that numcpus will be 0 if it is not provided. It looks like you may have set it to be 0 if it's not present and the comment above means that you will set it to 1. If that's the case I think the comments could use some cleanup.